provider/export/src/lib.rs

```diff
@@ -115,15 +116,38 @@ use std::hash::Hash;
 ///     )
 ///     .unwrap();
 /// ```
-#[derive(Debug, Clone)]
+#[derive(Clone)]
 pub struct ExportDriver {
     markers: Option<HashSet<DataMarkerInfo>>,
     requested_families: HashMap<DataLocale, DataLocaleFamilyAnnotations>,
+    #[allow(clippy::type_complexity)] // HashMap of cloneable dyns
+    attributes_filters: HashMap<String, Arc<Box<dyn DataMarkerAttributesFilter>>>,
     fallbacker: LocaleFallbacker,
     include_full: bool,
     deduplication_strategy: DeduplicationStrategy,
-    additional_collations: HashSet<String>,
-    segmenter_models: Vec<String>,
+}
+
+/// TODO
+pub trait DataMarkerAttributesFilter:
+    Fn(&DataMarkerAttributes) -> bool + Send + Sync + 'static
+{
+}
+impl<T> DataMarkerAttributesFilter for T where
+    T: Fn(&DataMarkerAttributes) -> bool + Send + Sync + 'static
+{
+}
```

Comment:
Suggestion:

```suggestion
/// A filtering predicate for [`DataMarkerAttributes`].
pub trait DataMarkerAttributesFilter: Send + Sync + 'static {
    /// Returns `true` if the `attributes` should be included.
    fn predicate(attributes: &DataMarkerAttributes) -> bool;
}

impl<T> DataMarkerAttributesFilter for T where
    T: Fn(&DataMarkerAttributes) -> bool + Send + Sync + 'static
{
    fn predicate(attributes: &DataMarkerAttributes) -> bool {
        T(attributes)
    }
}
```

Or, remove the trait and make the public function take the `Fn` directly